plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/InfluxMetadata.java

```diff
@@ -0,0 +1,240 @@
+    public InfluxTableHandle getTableHandle(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        if (!listSchemaNames(session).contains(schemaTableName.getSchemaName())) {
+            return null;
+        }
+
+        Optional<InfluxTable> table = client.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+        if (table.isEmpty()) {
+            return null;
+        }
+
+        return new InfluxTableHandle(schemaTableName.getSchemaName(), schemaTableName.getTableName(), TupleDomain.all());
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return getTableMetadata(((InfluxTableHandle) tableHandle).toSchemaTableName());
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        InfluxTableHandle influxTableHandle = (InfluxTableHandle) tableHandle;
+
+        Optional<InfluxTable> table = client.getTable(influxTableHandle.getSchemaName(), influxTableHandle.getTableName());
+        if (table.isEmpty()) {
+            throw new TableNotFoundException(influxTableHandle.toSchemaTableName());
+        }
+
+        ImmutableMap.Builder<String, ColumnHandle> columnHandles = ImmutableMap.builder();
+        for (InfluxColumn column : table.get().getColumns()) {
+            columnHandles.put(column.getName(), new InfluxColumnHandle(column.getName(), toTrinoType(column.getType()).orElse(null)));
+        }
+        return columnHandles.buildOrThrow();
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        return ((InfluxColumnHandle) columnHandle).getColumnMetadata();
+    }
+
+    @Override
+    public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        requireNonNull(prefix, "prefix is null");
+        ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
+        for (SchemaTableName tableName : listTables(session, prefix.getSchema())) {
+            if (!prefix.matches(tableName)) {
+                continue;
+            }
+            ConnectorTableMetadata tableMetadata = getTableMetadata(tableName);
+            if (tableMetadata != null) {
+                columns.put(tableName, tableMetadata.getColumns());
+            }
+        }
+        return columns.buildOrThrow().entrySet().stream()
+                .map(entry -> TableColumnsMetadata.forTable(entry.getKey(), entry.getValue())).iterator();
+    }
+
+    @Override
+    public Optional<LimitApplicationResult<ConnectorTableHandle>> applyLimit(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            long limit)
+    {
+        InfluxTableHandle tableHandle = (InfluxTableHandle) handle;
```

Comment:
Shouldn't we add additional checks if `InfluxTableHandle` already has limit and if is less than this value